### DISCUSSION TITLE
How can I override the standard DisplayDriver

### DISCUSSION THREAD

**Discussion by PiemP** [OrchardCMS/OrchardCore#12148] [Q&A] [answered]
Good evening, 

I have to override a DisplayDriver because the default one do some validation that is strange for my point of view but I don't know if is correct or not. 

I have tried to add this code to my module in the 'Startup.cs' using a my version of 'LinkMenuItemPartDisplayDriver':  

```
public override void ConfigureServices(IServiceCollection services)
        {
            // LinkMenuItemPart
            services.AddContentPart<LinkMenuItemPart>()
                .UseDisplayDriver<LinkMenuItemPartDisplayDriver>();

            services.AddScoped<IDataMigration, Migrations>();
        }
```

but when I start to test it, I have seen that this code register twice 'LinkMenuItemPart'. How can I do this in the right way?

Thank you

**Comment by PiemP** [comment]
I have solved in this way:

```
public override void ConfigureServices(IServiceCollection services)
        {
            // LinkMenuItemPart
            var linkmenuitempart = new ContentPartOptionBuilder(services, typeof(LinkMenuItemPart));
            linkmenuitempart.RemoveDisplayDriver<OrchardCore.Menu.Drivers.LinkMenuItemPartDisplayDriver>();
            linkmenuitempart.UseDisplayDriver<LinkMenuItemPartDisplayDriver>();

            services.AddScoped<IDataMigration, Migrations>();
        }
```

is it correct or exist a "best way"?

**Comment by ns8482e** [comment]
Ideally you can safely call multiple times
```
services.AddContentPart<LinkMenuItemPart>()
                .UseDisplayDriver<LinkMenuItemPartDisplayDriver>();
```

**Reply by PiemP** [reply]
I'm using 1.4.0 nuget package and if I try to use your code, it register twice `LinkMenuItemPart` and I see its fields twice in the backend interfaces. 

Another problem is about `UseDisplayDriver`: doesn't override the default behavior but add your customization to it. For this reason I have added the call to `RemoveDisplayDriver`.

Do I made something wrong when I try to override the default behavior?

I don't understand what I'm doing wrong with `UseDisplayDriver`: my code customization is the same of the default ones without a check in the `UpdateAsync` function. 

Thank you.

**Reply by ns8482e** [reply]
This code registers part and driver, it doesn't update definitions. If you see fields more than once then it's your migration causing the issue

**Reply by PiemP** [reply]
The fields duplication happen if I change my code to yours on an already initialized site... no new migrations... could be related on the nature of the `LinkMenuItemPart`?

**Reply by PiemP** [reply, marked as answer]
Ok, I have found the problem. Is related to this:

```
When registering a custom display mode or editor driver you must alter the registrations for existing drivers. 
You should also take a dependency in your modules `Manifest.cs` on the module that the fields reside in. 
This will make your modules Startup.cs run later, and allow your registrations to override the original modules
```

https://docs.orchardcore.net/en/latest/docs/reference/modules/ContentFields/#customising-display-driver-registration
